upload/swift.go

```diff
@@ -0,0 +1,85 @@
+package upload
+
+import (
+	"io/ioutil"
+	"log"
+	"path"
+	"path/filepath"
+
+	"github.com/ncw/swift"
+
+	"data-processing-spark-submit/utils"
+)
+
+type (
+	Swift struct {
+		StorageI
+		c *swift.Connection
+	}
+
+	SwiftConf struct {
+		UserName string `ini:"user_name"`
+		Password string `ini:"password"`
+		AuthURL  string `ini:"auth_url"`
+		Domain   string `ini:"domain"`
+		Region   string `ini:"region"`
+	}
+)
+
+//init storage
+func NewSwift(conf *SwiftConf) (*Swift, error) {
+	c := &swift.Connection{
+		UserName: conf.UserName,
+		ApiKey:   conf.Password,
+		AuthUrl:  conf.AuthURL,
+		Domain:   conf.Domain,
+		Region:   conf.Region,
+	}
+	return &Swift{
+		c: c,
+	}, nil
+}
+
+func (s *Swift) Upload(source, dest string) error {
+
+	if filepath.Ext(source) != "" {
+		//is dir
+		if err := s.Put(source, dest); err != nil {
+			return err
```

Comment:
Log this error. Even if it's logged higher in the call chain, you can add more valuable information here.